sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/GlobalEndpointManager.java

```diff
@@ -310,6 +323,7 @@ private Mono<DatabaseAccount> getDatabaseAccountAsync(URI serviceEndpoint) {
             .doOnNext(databaseAccount -> {
                 if(databaseAccount != null) {
                     this.latestDatabaseAccount = databaseAccount;
+                    this.setLatestDatabaseRefreshError(null);
```

Comment:
I am wondering whether we should track the exception itself instead of the error message, and then later when new RunTimeException(message, throwable)

Reply:
Ok. I made change to track exception, which I think addresses also Kushagra concern?

Reply:
Not really - but with my changes it should be fine.

Reply:
In RxDocumentClientImp - the important part is that the throwable in both logging and when wrapping in RuntimException should be passed in as cause - doing that with null is fine. For consistency you also have to use a snapshot - otherwise the value might have changed between logging and throwing inner exception